api/controllers/items.go

```diff
@@ -0,0 +1,47 @@
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/PedroChaparro/loomies-backend/interfaces"
+	"github.com/PedroChaparro/loomies-backend/models"
+	"github.com/gin-gonic/gin"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+func HandleGetItems(c *gin.Context) {
+	userid, _ := c.Get("userid")
+	user, err := models.GetUserById(userid.(string))
+
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			c.AbortWithStatusJSON(http.StatusNotFound, gin.H{"message": "User was not found"})
+			return
+		} else {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"message": "Internal server error"})
+			return
+		}
+	}
+
+	user_items := []interfaces.GetItem{}
+
+	for i := 0; i < len(user.Items); i++ {
```

Comment:
You can use an aggregation query instead of this for loop. Refer to the GetNearGyms function in the models/zones.go file or the GetBaseLoomies in the zones/loomies.go one for some examples.